### PR TITLE
Allow proxy endpoints to disable sub spans around downstream calls and trace header propagation

### PR DESCRIPTION
**Pull request by rabeyta**
Currently all proxy endpoints create a subspan around the downstream call and propagate all trace headers by default on every request

This PR adds the following two abilities
* allows an endpoint to configure a request to turn off trace header propagation 
* allows an endpoint to configure a request to not create a subspan around the downstream call

Default for these features is existing behavior, so this is a backwards compatible enhancement

### CODE REVIEW

File: riposte-core/src/test/java/com/nike/riposte/server/http/ProxyRouterEndpointTest.java [hunk 1]
```diff
@@ -43,4 +46,55 @@ public void isValidateRequestContent_returnsFalse() {
...
+
+    @Test
+    public void downstreamRequestFirstChunkInfo_allowsOverrideOfRelaxHttpsValidationFlag() {
+        // when
+        DownstreamRequestFirstChunkInfo downstreamRequestFirstChunkInfo =
+                new DownstreamRequestFirstChunkInfo("localhost", 8080, true, null)
+                        .withRelaxedHttpsValidation(true);
+
+        // then
+        assertThat(downstreamRequestFirstChunkInfo.relaxedHttpsValidation).isTrue();
```
[the remark this answers is not in the document]
Comment (rabeyta) [reply riposte-core/src/test/java/com/nike/riposte/server/http/ProxyRouterEndpointTest.java]: good call. i've refactored the tests i added to use data providers

File: riposte-core/src/test/java/com/nike/riposte/server/http/ProxyRouterEndpointTest.java [hunk 2]
```diff
@@ -43,4 +46,55 @@ public void isValidateRequestContent_returnsFalse() {
...
+
+    @Test
+    public void downstreamRequestFirstChunkInfo_allowsOverrideOfAddTracingFlag() {
+        // when
+        DownstreamRequestFirstChunkInfo downstreamRequestFirstChunkInfo =
+                new DownstreamRequestFirstChunkInfo("localhost", 8080, true, null)
+                .withAddTracingHeadersToDownstreamCall(false);
+
+        // then
+        assertThat(downstreamRequestFirstChunkInfo.addTracingHeadersToDownstreamCall).isFalse();
```
[the remark this answers is not in the document]
Comment (rabeyta) [reply riposte-core/src/test/java/com/nike/riposte/server/http/ProxyRouterEndpointTest.java]: done!

File: riposte-core/src/test/java/com/nike/riposte/server/http/ProxyRouterEndpointTest.java [hunk 3]
```diff
@@ -43,4 +46,55 @@ public void isValidateRequestContent_returnsFalse() {
...
+
+    @Test
+    public void downstreamRequestFirstChunkInfo_allowsOverrideOfPerformSubspanOnDownstreamCall() {
+        // when
+        DownstreamRequestFirstChunkInfo downstreamRequestFirstChunkInfo =
+                new DownstreamRequestFirstChunkInfo("localhost", 8080, true, null)
+                        .withPerformSubSpanAroundDownstreamCall(false);
+
+        // then
+        assertThat(downstreamRequestFirstChunkInfo.performSubSpanAroundDownstreamCall).isFalse();
```
[the remark this answers is not in the document]
Comment (rabeyta) [reply riposte-core/src/test/java/com/nike/riposte/server/http/ProxyRouterEndpointTest.java]: done!

**Comment by nicmunroe** [comment]
Looks good - thank you!

**Pull request merged**